waspc/data/Generator/templates/react-app/src/api.ts

```diff
@@ -1,29 +1,40 @@
 import axios, { type AxiosError } from 'axios'
+import EventEmitter from 'events'
+
 import config from './config'
 import { storage } from './storage'
 
 const api = axios.create({
   baseURL: config.apiUrl,
 })
 
+// Used to allow API clients to register for auth token change events.
+export const events = new EventEmitter()
```

Comment:
Yeah, I can see what that would look like 👍🏻

Reply:
Ok, if it is too complex feel free to punt on it though.

Reply:
I think the typed ones may be, but I will at least make them string constants 👍🏻

Reply:
I would consider wrapping the `EventEmitter` to:
- allow us to type allowed events
- abstract away that we are using `EventEmitter` 

Something like this:
```ts
export const EventType = {
  SET_AUTH_TOKEN: 'authToken.set',
  CLEAR_AUTH_TOKEN: 'authToken.clear',
} as const;

// 'authToken.set' | 'authToken.clear'
type EventType = typeof EventType[keyof typeof EventType]

function createPubSub() {
  const events = new EventEmitter();

 return {
    emit(event: EventType): void {
      events.emit(event);
    },
    on(event: EventType, handler: () => void): void {
      events.on(event, handler) 
   }
  }
}

export const events = createPubSub();
```

Reply:
Interesting -> although this look exactly the same as event emitter, interface wise? So I am not sure we got a lot here regarding abstracting away, it is the same interface really.

I would say the only thing we got are types. But isn't there a typed event emitter out there in the world, do we really have to write our own? Should we maybe just name it TypedEventEmitter then and have that in the codebase?

Reply:
> Interesting -> although this look exactly the same as event emitter, interface wise? So I am not sure we got a lot here regarding abstracting away, it is the same interface really.

If we wanted to swap out the lib we are using, it would be easier :)

This `events` lib is older and it wasn't written with Typescript in mind. Maybe we can find a different one? Or be without types, that's okay with me since it's a limited use case.

Reply:
I think this events lib comes is builtin? But you know what, I am ok with any approach here, whatever you choose, including what you suggested above.

Reply:
I think this doesn't work since `EventEmitter` is not available in the browser. It _seems_ to work in our code because we installed `@types/node`, but this code shouldn't work in any browser.

I'll see if we can use a different library/API